tapalcatl_server/server.go

```diff
@@ -333,15 +331,11 @@ func main() {
 	// metrics writer configuration
 	var mw metricsWriter
 	if metricsStatsdAddr != "" {
-		if metricsInterval <= 0 {
-			logger.Fatalf("ERROR: metrics interval should be greater than 0")
-		}
 		udpAddr, err := net.ResolveUDPAddr("udp4", metricsStatsdAddr)
 		if err != nil {
 			logger.Fatalf("ERROR: Invalid metricsstatsdaddr %s: %s", metricsStatsdAddr, err)
 		}
-		metricsIntervalDuration := time.Second * time.Duration(metricsInterval)
-		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, metricsIntervalDuration)
+		mw = NewStatsdMetricsWriter(udpAddr, metricsStatsdPrefix, logger)
```

Comment:
I've had issues in the past, in a different environment and runtime - not Go, with programs not cleaning up the threads that they've spawned. The effect of that was that a problem in one thread caused the whole system to deadlock and stop accepting requests, rather than to die cleanly and be restarted.

Since then, I've always been a bit paranoid about making sure that threads are correctly managed, and get set shutdown signals. In Go, that means closing the `chan` that the goroutine is  `range`ing over. However, in Go a goroutine isn't a thread, and I haven't fully internalised all the logic yet about the rules when they'll get automatically stopped, or when channels will get automatically closed, etc...

All of which is a long way of saying: I thought there might be an issue here with the metrics writer goroutine outliving the main goroutine, but I was wrong. :wink: